tests/scripts/check_names.py

```diff
@@ -317,25 +328,11 @@ def get_files(self, include_wildcards, exclude_wildcards):
         """
         accumulator = set()
 
-        # exclude_wildcards may be None. Also, consider the global exclusions.
-        exclude_wildcards = (exclude_wildcards or []) + self.excluded_files
-
-        # Internal function to hit the memoisation cache or add to it the result
-        # of a glob operation. Used both for inclusion and exclusion since the
-        # only difference between them is whether they perform set union or
-        # difference on the return value of this function.
-        def hit_cache(wildcard):
-            if wildcard not in self.files:
-                self.files[wildcard] = set(glob.glob(wildcard, recursive=True))
-            return self.files[wildcard]
-
         for include_wildcard in include_wildcards:
-            accumulator = accumulator.union(hit_cache(include_wildcard))
-
-        for exclude_wildcard in exclude_wildcards:
-            accumulator = accumulator.difference(hit_cache(exclude_wildcard))
+            accumulator = accumulator.union(glob.iglob(include_wildcard))
```

Comment:
Note: the previous version used `glob`, this one uses `iglob` - it matches the same files but just returns an iterator, so that's good. (For a moment I thought the difference was the same as between `-name` in `-iname` with `find` and was confused.)